treon/treon.py

```diff
@@ -106,6 +109,13 @@ def print_test_collection(notebooks):
     LOG.debug(message)
 
 
+def filter_results(results, args):
+    for exclude_str in args['--exclude']:
+        exclude_abs = os.path.abspath(os.path.expanduser(exclude_str))
+        results = [file for file in results if not os.path.abspath(file).startswith(exclude_abs)]
```

Comment:
I just found a problem with this, 

These are all the files I am testing normally,
```
(treon4) [amitrathi@MacBook-Pro-3 treon ((065c2f2...))]$ treon  ~/workspace/PythonDataScienceHandbook 
Executing treon version 0.1.2
Recursively scanning /Users/amitrathi/workspace/PythonDataScienceHandbook for notebooks...
Triggered test for /Users/amitrathi/workspace/PythonDataScienceHandbook/nb2/05.08-V2-Random-Forests.ipynb
Triggered test for /Users/amitrathi/workspace/PythonDataScienceHandbook/nb2/05.04-V2-Feature-Engineering.ipynb
Triggered test for /Users/amitrathi/workspace/PythonDataScienceHandbook/notebooks2/04.13v3.ipynb
Triggered test for /Users/amitrathi/workspace/PythonDataScienceHandbook/notebooks2/04.13v2.ipynb
Triggered test for /Users/amitrathi/workspace/PythonDataScienceHandbook/notebooks/03.10-Working-With-Strings.ipynb
Triggered test for /Users/amitrathi/workspace/PythonDataScienceHandbook/notebooks/05.09-Principal-Component-Analysis.ipynb
Triggered test for /Users/amitrathi/workspace/PythonDataScienceHandbook/notebooks/03.06-Concat-And-Append.ipynb
Triggered test for /Users/amitrathi/workspace/PythonDataScienceHandbook/notebooks/02.05-Computation-on-arrays-broadcasting.ipynb
Triggered test for /Users/amitrathi/workspace/PythonDataScienceHandbook/notebooks/05.05-Naive-Bayes.ipynb
Triggered test for /Users/amitrathi/workspace/PythonDataScienceHandbook/notebooks/04.14-Visualization-With-Seaborn.ipynb
```


Now when I try to ignore `......./notebooks` folder `......./notebooks2` is also ignored because of the startswith condition.
```
(treon4) [amitrathi@MacBook-Pro-3 treon ((065c2f2...))]$ treon ~/workspace/PythonDataScienceHandbook --exclude=~/workspace/PythonDataScienceHandbook/notebooks/ 
Executing treon version 0.1.2
Recursively scanning /Users/amitrathi/workspace/PythonDataScienceHandbook for notebooks...
Triggered test for /Users/amitrathi/workspace/PythonDataScienceHandbook/nb2/05.08-V2-Random-Forests.ipynb
Triggered test for /Users/amitrathi/workspace/PythonDataScienceHandbook/nb2/05.02-V2-Introducing-Scikit-Learn.ipynb
Triggered test for /Users/amitrathi/workspace/PythonDataScienceHandbook/nb2/05.04-V2-Feature-Engineering.ipynb
Triggered test for /Users/amitrathi/workspace/PythonDataScienceHandbook/nb2/03.07-Merge-and-Join.ipynb
Triggered test for /Users/amitrathi/workspace/PythonDataScienceHandbook/nb3/01.00-IPython-Beyond-Normal-Python.ipynb
```

Reply:
IMO there are 2 ways to solve this,

1. For each exclude path list all ipynb files under it (if path is directory) and put them in exclude set. Then exclude any file that is in the exclude set.

2. For each file check whether it is equal to exclude string OR (startswith exclude string AND exclude string has slash after the match).

Option 2 might be quick but dirty. Option 1 might be clean but inefficient. I leave it upto you.

Reply:
What do you suggest?
This is a feature, I guess. Only way to avoid this is to restrict to valid directory names and valid file paths. But this can be cumbersome as well. 
I am fine with both.

Reply:
>Only way to avoid this is to restrict to valid directory names and valid file paths

Option 2 above would be easy to implement. We wouldn't even need to check valid paths.

Reply:
I edited option 2 a bit below -

For each file check whether it is equal to exclude string OR (file startswith exclude string AND filename has slash after the match).

Reply:
some use the following check:
file_path.startswith(os.path.abspath(dir_path)+os.sep)

Reply:
Yep, something like that should do it.